larq_compute_engine/core/bgemm_kernels_ruy.h

```diff
@@ -98,6 +98,8 @@ struct BgemmKernel<ruy::Path::kStandardCpp, LhsScalar, RhsScalar, DstScalar,
         if (spec.fused_add) {
           dst_val += spec.fused_add[i];
         }
+        dst_val = std::min<DstScalar>(dst_val, spec.clamp_max);
+        dst_val = std::max<DstScalar>(dst_val, spec.clamp_min);
```

Comment:
This would now calculate `Conv -> Mul -> Add -> Relu`, but for us to be useful in practice we'd want to do `Conv -> Relu -> Mul -> Add` or am I missing something?

Do you think it would make sense to split `fused_add` and `fused_multiply` into the backtransformation to {-1, +1} space and the floating fused multiplier + bias?

Reply:
So i'd be in favor of having *two* multiply and add inputs. Maybe something like `{pre,post}_activation_{multiply,add}`.
The reason is that between Conv and Relu, we need a multiply and add *anyway* for the backtransformation. So we might as well support custom multiply and add values since the number of computational ops stays the same.
The multiply and add *after* the Relu are really extra, and the TF Lite conv also does not fuse them. Now that we have the MLIR converter, that'll be easy to add. I'll try to make a PR soon.